Assets/MixedRealityToolkit/_Core/Devices/MixedReality/WindowsMixedRealityController.cs

```diff
@@ -1,39 +1,251 @@
 ﻿// Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License. See LICENSE in the project root for license information.
 
-using System.Collections.Generic;
 using Microsoft.MixedReality.Toolkit.Internal.Definitions.Devices;
+using Microsoft.MixedReality.Toolkit.Internal.Definitions.InputSystem;
 using Microsoft.MixedReality.Toolkit.Internal.Definitions.Utilities;
+using Microsoft.MixedReality.Toolkit.Internal.Extensions;
+using Microsoft.MixedReality.Toolkit.Internal.Interfaces;
 using Microsoft.MixedReality.Toolkit.Internal.Interfaces.InputSystem;
+using Microsoft.MixedReality.Toolkit.Internal.Utilities;
+using System;
+using System.Collections.Generic;
+using UnityEngine;
+using UnityEngine.XR.WSA.Input;
 
 namespace Microsoft.MixedReality.Toolkit.Internal.Devices.WindowsMixedReality
 {
     public struct WindowsMixedRealityController : IMixedRealityController
     {
-        public WindowsMixedRealityController(ControllerState controllerState, Handedness controllerHandedness, IMixedRealityInputSource inputSource, Dictionary<DeviceInputType, InteractionMapping> interactions) : this()
+        #region Private properties
+
+        private bool controllerTracked;
+        private Vector3 controllerPosition;
+        private Vector3 pointerPosition;
+        private Vector3 gripPosition;
+        private Quaternion controllerRotation;
+        private Quaternion pointerRotation;
+        private Quaternion gripRotation;
+
+        #endregion Private properties
+
+        public WindowsMixedRealityController(ControllerState controllerState, Handedness controllerHandedness, IMixedRealityInputSource inputSource, Dictionary<DeviceInputType, InteractionMapping> interactions = null) : this()
         {
             ControllerState = controllerState;
             ControllerHandedness = controllerHandedness;
             InputSource = inputSource;
-            Interactions = interactions;
+            Interactions = interactions ?? new Dictionary<DeviceInputType, InteractionMapping>();
+
+            controllerTracked = false;
+            controllerPosition = pointerPosition = gripPosition = Vector3.zero;
+            controllerRotation = pointerRotation = gripRotation = Quaternion.identity;
         }
 
-        public ControllerState ControllerState { get; }
+        #region IMixedRealityController Interface Members
+
+        public ControllerState ControllerState { get; private set; }
 
         public Handedness ControllerHandedness { get; }
 
-        public IMixedRealityInputSource InputSource { get; }
+        public IMixedRealityInputSource InputSource { get; private set; }
+
+        public Dictionary<DeviceInputType, InteractionMapping> Interactions { get; private set; }
+
+        public void SetupInputSource<T>(T state)
+        {
+            InteractionSourceState interactionSourceState = CheckIfValidInteractionSourceState(state);
```

Comment:
No behavior change if CheckIfValid fails?

Reply:
Good point, will review

Reply:
Double checked.  The function will only ever return a valid InteractionSource state.  If the type of the parameter passed in is not an InteractionSourceState, it will throw an ArgumentOutOfRangeException.

Don't think we need anything more there really.  In reality, the check isn't needed, but I like to walk on the safe side and be sure.